actionmailer/test/test_helper_test.rb

```diff
@@ -487,6 +487,68 @@ def test_assert_enqueued_email_with_with_no_block_with_parameterized_args
       end
     end
   end
+
+  def test_deliver_enqueued_emails_with_no_block
+    assert_nothing_raised do
+      silence_stream($stdout) do
+        TestHelperMailer.test.deliver_later
+        deliver_enqueued_emails
+      end
+    end
+
+    assert_emails(1)
```

Comment:
If you pass it a block, `assert_emails` delivers emails enqueued by the block. So to properly test this it needs to be called without a block.